ad9959/src/lib.rs

```diff
@@ -568,6 +609,58 @@ impl ProfileSerializer {
         }
     }
 
+    /// Update a number of channels with fully defined profile settings.
+    ///
+    /// # Args
+    /// * `channels` - A set of channels to apply the configuration to.
+    /// * `profile` - The complete DDS profile, which defines the frequency tuning word,
+    ///   amplitude control register & the phase offset word of the channels. Note that the ACR
+    ///   should be stored in the 3 LSB of the word. If amplitude scaling is to be used, the
+    ///   "Amplitude multiplier enable" bit must be set.
+    #[inline]
+    pub fn update_channels_with_profile(
+        &mut self,
+        channels: Channel,
+        profile: DdsProfile,
+    ) {
+        let csr = [self.mode as u8 | channels.bits()];
+        self.add_write(Register::CSR, &csr);
+        self.add_write(Register::CFTW0, &profile.ftw.to_be_bytes());
+        self.add_write(Register::CPOW0, &profile.pow.to_be_bytes());
+        self.add_write(Register::ACR, &profile.acr.to_be_bytes()[1..]);
+    }
+
+    /// Update the system clock configuration.
+    ///
+    /// # Args
+    /// * `reference_clock_frequency` - The reference clock frequency provided to the AD9959 core.
+    /// * `multiplier` - The frequency multiplier of the system clock. Must be 1 or 4-20.
+    #[inline]
+    pub fn update_system_clock(
+        &mut self,
+        reference_clock_frequency: f32,
+        multiplier: u8,
+    ) -> Result<f32, Error> {
+        let frequency = Ad9959::<()>::validate_system_clock(
```

Comment:
No. That's a layering violation. Don't do the conversions and these checks in the serializer but elsewhere.